lightning/src/chain/channelmonitor.rs

```diff
@@ -4095,6 +4095,7 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 						revocation_pubkey: broadcasted_holder_revokable_script.2,
 						channel_keys_id: self.channel_keys_id,
 						channel_value_satoshis: self.channel_value_satoshis,
+						delayed_payment_basepoint: self.onchain_tx_handler.signer.pubkeys().delayed_payment_basepoint,
```

Comment:
I included delayed_payment_basepoint in the descriptor, because we need it to construct PSBT and we don't want to extend the `SpendableOutputDescriptor::create_spendable_outputs_psbt` signature (see [comment](https://github.com/lightningdevkit/rust-lightning/pull/2761#discussion_r1409987188)). 

If we want to re-derive the keys in `create_spendable_outputs_psbt` then we just replace that argument with a node xpub, which is useful, because it still needs static payment basepoint for a StaticPaymentOutput. 

To not overload this function, we could have another struct that would implement a new trait called ChannelPublicKeysDeriver which can be asked to derive all types of channel public keys.

Reply:
Basically it is being able to use L1 signer. 

Here is an example interface for a signer https://github.com/lightsparkdev/lightspark-rs/blob/main/lightspark-remote-signing/src/signer.rs#L160. 

Inputs:
```
message: Vec<u8>,
derivation_path: String,
add_tweak: Option<Vec<u8>>,
mul_tweak: Option<Vec<u8>>,
```

So it can do some BIP32 derivations, but not L2 derivations. I am not sure why it is this way, but I it is conceivable that it helps with compatibility with different non-custodial signers.

Reply:
It looks much nicer with ChannelPublicKeys injection, but I left it to be optional for backward compatibility.
I also added PSBT input generation for a StaticPaymentOutput, where I make an assumption that per_commitment_point is not used (and hence key tweak is not needed). I don't know if I could assert this assumption in the code.
Lastly, moved derive add tweak to the Basepoint implementation in channel_keys.rs.
Please let me know if this looks correct or needs some more work.